docs/guide/nameservice/03-types.md

```diff
@@ -1,40 +1,83 @@
-* `x/nameservice/client/cli/query.go`: registering CLI commands.
-* `x/nameservice/client/cli/query_whois_test.go`: tests for CLI commands.
-* `x/nameservice/types/keys.go`: a string as a prefix in the key used to store whois information in the state.
-* `x/nameservice/genesis.go`: logic for exporting and exporting the state.
-* `x/nameservice/types/genesis.go`: logic for validating the genesis file.
-* `x/nameservice/module.go`: registering gRPC gateway routes.
-
-`SetWhois` in the `keeper` package uses a key-value store using a prefix for the whois type (`Whois-value-`) encodes the `Whois` type (that is generated from a protocol buffer definition) and uses `store.Set` method to write a Whois into the store.
-
-`GetWhois` selects a store using the whois prefix, and uses `store.Get` to fetch a `whois` with a particular index.
+where:
+
+- whois is the type
+- name is the name the user sets
+- value is the name that name resolves to
+- price is the bid 
+- `--no-message` flag skips message creation 
+    By default, generic CRUD messages are scaffolded. However, you've already created messages specifically for this blockchain, so you can skip message creation with the `--no-message` flag.
+
+The `starport scaffold map whois name value price --no-message` command created and modified several files:
+
+* `proto/nameservice/whois.proto`
+    Defines the `Whois` type as a proto message.
+* `proto/nameservice/query.proto`
+    * Queries to get data from the blockchain. 
+    * Define queries as proto messages.
+    * Register the messages in the `Query` service.
+* `proto/nameservice/genesis.proto`
+    A type for exporting the state of the blockchain, for example, during software upgrades.
+* `x/nameservice/keeper/grpc_query_whois.go`
+    Keeper methods to query the blockchain.
+* `x/nameservice/keeper/grpc_query_whois_test.go`
+    Tests for query keeper methods.
+* `x/nameservice/keeper/whois.go`
+    Keeper methods to get, set, and remove whois information from the store.
+* `x/nameservice/keeper/whois_test.go`
+    Tests for keeper methods.
+* `x/nameservice/client/cli/query_whois.go`
+    CLI commands for querying the blockchain.
+* `x/nameservice/client/cli/query.go`
+    Registers the CLI commands.
+* `x/nameservice/client/cli/query_whois_test.go`
+    Tests for CLI commands.
+* `x/nameservice/types/keys.go`
+    String prefix in the key to store whois information in the state.
+* `x/nameservice/genesis.go`
+    Logic for exporting and exporting the state.
+* `x/nameservice/types/genesis.go`
+    Logic for validating the genesis file.
+* `x/nameservice/module.go`
+    Registers gRPC gateway routes.
+
+## Keeper Package
+
+In the `x/nameservice/keeper/whois.go` file, take at a look at the keeper package. 
+
+- `SetWhois` uses a key-value store with a prefix for the `Whois` type and uses a `store.Set` method to write a `Whois` into the store.
+
+<!-- where is this? teach me please 
```

Comment:
not sure what this existing content is talking about, teach me?

Reply:
In `x/nameservice/keeper/whois.go`.